compiler/noirc_evaluator/src/errors.rs

```diff
@@ -171,6 +174,17 @@ impl RuntimeError {
                     location.span,
                 )
             }
+            RuntimeError::UnconstrainedSliceReturnToConstrained { .. } => {
+                let primary_message = self.to_string();
+                let location =
+                    self.call_stack().back().expect("Expected RuntimeError to have a location");
+
+                Diagnostic::simple_error(
+                    primary_message,
+                    "If attempting to return a `Vec` type, `Vec` contains a slice internally.".to_string(),
```

Comment:
I'm not a fan of these hyper-specific taglines which might now apply to the actual cause of the error from the user. If we're not specifically checking for Vec, we shouldn't mention it in the error message imo. Otherwise it makes the message sound more confusing / imprecise.

Reply:
Yeah that is a fair point. I tried to mitigate with the language "If attempting". I did it this way as this issues an error when we do codegen on the call which isn't super clear about the return type itself. I figured if a user wrote a struct with a slice in it they would know, but for `Vec` they may be unsure as it is one of our stdlib types.